File: GestorActividades.Services/Implementation/ActivityService.cs
using GestorActividades.Infrastructure;
using GestorActividades.Infrastructure.Models;
using GestorActividades.Services.Validation;
using System;
using System.Linq;

namespace GestorActividades.Services
{
    public class ActivityService : ServiceBase, IActivityService
    {
        public ResponseDto<Activity> AddActivity(Activity Activity)
        {
            var response = new ResponseDto<Activity> { StatusCode = StatusCode.Error };

            var validation = ModelHelperValidator.ValidateComponentModel(Activity);

            if (!validation.Data)
            {
                response.StatusCode = StatusCode.Error;
                response.StatusMessage = validation.StatusMessage.Trim();
                return response;
            }

            using (var unitOfWork = UnitOfWorkFactory.GetUnitOfWork())
            {
                var repository = unitOfWork.GetGenericRepository<Activity>();
                var repositoryTeam = unitOfWork.GetGenericRepository<Team>();

                Activity.Description = Activity.Description.Trim();                
                Activity.CreatedDt = DateTime.Now;

                if (!repositoryTeam.GetAll().Any(x => x.TeamId == Activity.TeamId))
                {
                    response.StatusCode = StatusCode.Error;
                    response.StatusMessage = "The Team doesn't exist in the system.";
                    return response;
                }

                repository.InsertAndSave(Activity);


                response.StatusCode = StatusCode.Successful;
                response.Data = Activity;
            }

            return response;

        }

        public ResponseDto<Activity> GetActivityById(int id)
        {
            var response = new ResponseDto<Activity>(StatusCode.Successful);

            using (var unitOfWork = UnitOfWorkFactory.GetUnitOfWork())
            {
                var repo = unitOfWork.GetGenericRepository<Activity>();

                response.Data = repo.GetAll().FirstOrDefault(x => x.ActivityId == id);

                if (response.Data == null)
                {
                    response.StatusCode = StatusCode.Error;
                }

                return response;
            }
        }

        public ResponseDto<Activity> UpdateActivity(Activity Activity)
        {
            var response = new ResponseDto<Activity>(StatusCode.Successful);

            using (var unitOfWork = UnitOfWorkFactory.GetUnitOfWork())
            {
                var repo = unitOfWork.GetGenericRepository<Activity>();
                var repositoryTeam = unitOfWork.GetGenericRepository<Team>();

                IQueryable<Activity> activities = repo.GetAll();

                response.Data = activities.FirstOrDefault(x => x.ActivityId == Activity.ActivityId);

                if (response.Data == null)
                {
                    response.StatusCode = StatusCode.Warning;
                    return response;
                }

                var validation = ModelHelperValidator.ValidateComponentModel(Activity);

                if (!validation.Data)
                {
                    response.StatusCode = StatusCode.Error;
                    response.StatusMessage = validation.StatusMessage.Trim();
                    return response;
                }

                if (!repositoryTeam.GetAll().Any(x => x.TeamId == Activity.TeamId))
                {
                    response.StatusCode = StatusCode.Error;
                    response.StatusMessage = "The team doesn't exist in the system.";
                    return response;
                }

                response.Data.ActivityId = Activity.ActivityId;
                response.Data.Description = Activity.Description;
                response.Data.StartDate = Activity.StartDate;
                response.Data.DueDate = Activity.DueDate;

                repo.UpdateAndSave(response.Data);

                return response;
            }
        }

        public ResponseDto<bool> DeleteActivity(int id)
        {
            var response = new ResponseDto<bool>(StatusCode.Successful);

            using (var unitOfWork = UnitOfWorkFactory.GetUnitOfWork())
            {
                var repo = unitOfWork.GetGenericRepository<Activity>();

                var Activity = repo.GetAll().FirstOrDefault(x => x.ActivityId == id);

                if (Activity == null)
                {
                    response.StatusCode = StatusCode.Error;
                    return response;
                }

                repo.Delete(Activity);
                repo.Save();

                return response;
            }
        }
    }
}

File: GestorActividades/activity-manager/src/app/services/activity.service.ts
import { Injectable } from '@angular/core';
import { Activity } from "./activity";
import { Observable, of } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ActivityService {

  activities: Activity[] = [
    { ActivityId: 1, Title: "Activity 1", Description: "Description Activity 1", StartDate: new Date(), DueDate: new Date(), CreatedDt: new Date(), Status: 1 },
    { ActivityId: 2, Title: "Activity 2", Description: "Description Activity 2", StartDate: new Date(), DueDate: new Date(), CreatedDt: new Date(), Status: 1 },
    { ActivityId: 3, Title: "Activity 3", Description: "Description Activity 3", StartDate: new Date(), DueDate: new Date(), CreatedDt: new Date(), Status: 3 },
    { ActivityId: 4, Title: "Activity 4", Description: "Description Activity 4", StartDate: new Date(), DueDate: new Date(), CreatedDt: new Date(), Status: 3 },
    { ActivityId: 5, Title: "Activity 5", Description: "Description Activity 5", StartDate: new Date(), DueDate: new Date(), CreatedDt: new Date(), Status: 2 },
    { ActivityId: 6, Title: "Activity 6", Description: "Description Activity 6", StartDate: new Date(), DueDate: new Date(), CreatedDt: new Date(), Status: 1 },
    { ActivityId: 7, Title: "Activity 7", Description: "Description Activity 7", StartDate: new Date(), DueDate: new Date(), CreatedDt: new Date(), Status: 3 }
  ];

  constructor() { }

  getActivities() : Observable<Activity[]>
  {
    return of(this.activities);
  }
}
